File: Solutions in Java/MajorityElement.java
/**
 * @author : Anmol Agrawal
 */
class Solution {
    public int majorityElement(int[] nums) {
        int m = nums.length / 2;
        Map<Integer, Integer> freq = new HashMap<>();
        for(int x : nums){
            freq.put(x, freq.getOrDefault(x, 0) + 1);
        }
        for(Map.Entry<Integer, Integer> entry : freq.entrySet()){
            if(entry.getValue() > m) return entry.getKey();
        }
        return 0;
    }
}

File: Solutions in Javascript/MajorityElement.js
/**
 * @author : Anmol Agrawal
 */
/**
 * @param {number[]} nums
 * @return {number}
 */
max = 0;
maxKey = 0;
var majorityElement = function(nums) {
    max = 0;
    maxKey = 0;
    let map = new Map()
    for(let i = 0; i < nums.length; i++){
        if (map.has(nums[i])) {
            map.set(nums[i], map.get(nums[i]) + 1);
        }
        else {
            map.set(nums[i], 1);
        }
    }
    map.forEach(logMapElements);
    return maxKey;
};
function logMapElements(value, key, map) {
    if(value > max){
        max = value;
        maxKey = key;
    }
}
